src/controllers/apps/ecommerce/coupon.controllers.js

```diff
@@ -19,6 +19,22 @@ const createCoupon = asyncHandler(async (req, res) => {
     expiryDate,
   } = req.body;
 
+  // Convert the provided start and expiry dates to JavaScript Date objects
+  const start = new Date(startDate);
+  const expiry = new Date(expiryDate);
+
+  // Check if the start date is in the future
+  // If the start date is in the past (i.e., before the current date and time), throw an error
+  if (start < new Date()) {
+    throw new ApiError(400, "Start date must be in the future");
+  }
+
+  // Check if the expiry date is in the future
+  // If the expiry date is in the past (i.e., before the current date and time), throw an error
+  if (expiry < new Date()) {
```

Comment:
Move this logic to the validators using express validator. Don't include validation logic in the controllers